pkg/reconciler/knservice/knservice_reconciler.go

```diff
@@ -0,0 +1,66 @@
+package knservice
+
+import (
+	"context"
+	"fmt"
+	"github.com/knative/pkg/kmp"
+	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("ServiceReconciler")
+
+type ServiceReconciler struct {
+	client client.Client
+}
+
+func NewServiceReconcile(client client.Client) *ServiceReconciler {
+	return &ServiceReconciler{
+		client: client,
+	}
+}
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Service resource
+// with the current status of the resource.
+func (c *ServiceReconciler) Reconcile(ctx context.Context, desiredService *knservingv1alpha1.Service) (*knservingv1alpha1.Service, error) {
+	service := &knservingv1alpha1.Service{}
+	err := c.client.Get(context.TODO(), types.NamespacedName{Name: desiredService.Name, Namespace: desiredService.Namespace}, service)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Creating service", "namespace", service.Namespace, "name", service.Name)
+		err = c.client.Create(context.TODO(), desiredService)
+		return desiredService, err
+	} else if err != nil {
+		return nil, err
```

Comment:
When do non 404 errors get logged e.g unauthorized

Reply:
the error will get logged in controller and record as an event for kfsvc.
```
	ksvc, err := serviceReconciler.Reconcile(context.TODO(), desiredService)
	if err != nil {
		log.Error(err, "Failed to reconcile knservice", "name", desiredService.Name)
		r.Recorder.Eventf(kfsvc, v1.EventTypeWarning, "InternalError", err.Error())
		return reconcile.Result{}, err
	}
```